hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/producer/kafka/KafkaMessagesAsyncProducerTest.groovy

```diff
@@ -0,0 +1,79 @@
+package pl.allegro.tech.hermes.frontend.producer.kafka
+
+import pl.allegro.tech.hermes.frontend.metric.CachedTopic
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessageProducer
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResults
+import pl.allegro.tech.hermes.frontend.publishing.PublishingCallback
+import pl.allegro.tech.hermes.frontend.publishing.message.JsonMessage
+import pl.allegro.tech.hermes.frontend.publishing.message.Message
+import spock.lang.Shared
+import spock.lang.Specification
+
+import static com.google.common.base.Charsets.UTF_8
+import static pl.allegro.tech.hermes.frontend.server.CachedTopicsTestHelper.cachedTopic
+
+class KafkaMessagesAsyncProducerTest extends Specification {
+
+    @Shared
+    CachedTopic cachedTopic = cachedTopic("group.test")
+
+    def "should successfully publish messages async if publishing single message on broker is available"() {
+        given:
+        List<Message> jsonMessages = (0..<4).collect {
+            new JsonMessage(it.toString(), "{\"data\":\"json\"}".getBytes(UTF_8), 1L, null)
+        }
+        BrokerMessageProducer brokerMessageProducer = TestBrokerMessageProducer.builder().throwException(false).build()
+        KafkaMessagesAsyncProducer kafkaMessagesAsyncProducer = new KafkaMessagesAsyncProducer(brokerMessageProducer, 4)
+
+        when:
+        BrokerMessagesProducingResults results = kafkaMessagesAsyncProducer.publishMessagesAsync(cachedTopic, jsonMessages, 100L)
+
+        then:
+        !results.isFailure()
+    }
+
+    def "should failed to publish messages async if exception is thrown by publishing single message on broker"() {
```

Comment:
`should failed to publish messages async if exception is thrown by publishing single message on broker` -> `should fail to publish messages broker is unavailable`

Reply:
but I want to show different cases here, when existing broker message producer throws an exception, this is unit test which depends on existing broker message producer and not on broker availability